android/src/BluetoothSensor.java

```diff
@@ -117,6 +117,32 @@ private void enableNotification(BluetoothGattCharacteristic c) {
     }
   }
 
+  /**
+   * Data in the characteristic has little endian byteorder.
+   * Lowest bit of flags indicates valid revs_per_sec reading.
+   * 0 Kelvin indicates invalid temperatures e.g 
+   * no temperature sensor present.
+  */
+  private void engineSensorDataToListeners(BluetoothGattCharacteristic c) {
+    final int flags = c.getIntValue(c.FORMAT_UINT8, 0);
+    final int cht_temp = c.getIntValue(c.FORMAT_UINT16, 1);
+    final int egt_temp = c.getIntValue(c.FORMAT_UINT16, 3);
+    final int outside_air_temperature = c.getIntValue(c.FORMAT_UINT16, 5);
+    if(outside_air_temperature != 0)
+      listener.onTemperature(outside_air_temperature);
+    final int pressure = c.getIntValue(c.FORMAT_UINT32, 7);
+    // Just guessing the sensor_noise_variance.
+    if(pressure != 0)
+      listener.onBarometricPressureSensor(c.getIntValue(c.FORMAT_UINT32, 7), 0.01f);
+    final int revs_per_sec = c.getIntValue(c.FORMAT_UINT16, 11);
```

Comment:
In your protocol design, if the flag isn't set, is the field present in the BLE packet? With a bogus value? Other BLE protocols I know just omit the field in that case, and then you shouldn't attempt to read it. The protocol is really our decision, I just want to point out that this needs to be an explicit decision, not an accident.

Reply:
I'm just starting the project. So the developed sensor box is not jet fully functional complete. Specifically when it comes to one safety feature. There are several severe injuries and fatalities each year, due to people starting their paramotors on the ground without having them secured properly. Last year someone got half his head chopped of, while his wife was watching. The throttle cable was stuck and he did not recognize. Engine started and went crazy on full throttle.
As a future feature I developed a security circuit, monitoring the engine for the first 20 seconds and do an immediate emergency shutdown of the engine if it revs up, by pulling the primary ignition coil to GND. This circuit board is close to ready for production.  The sensor box will signal a emergency shutdown by setting the flags bit to 0. After a little while it should go back to 1, so the engine can be restarted again. That's the plan with the flag :)